packages/chakra-ui/src/IconButton/index.d.ts

```diff
@@ -4,15 +4,15 @@ import { Omit } from "../common-types";
 
 type _ButtonProps = Omit<
   ButtonProps,
-  "loadingText" | "isFullWidth" | "leftIcon" | "rightIcon" | "iconSpacing"
+  "loadingText" | "isFullWidth" | "leftIcon" | "rightIcon" | "iconSpacing" | "children"
 >;
 
 interface IIconButton {
   /**
    * The icon to be used. Refer to the [Icons](https://chakra-ui.com/icon/) section
    * of the docs for the available icon options.
    */
-  icon?: string;
+  icon: string | JSX.Element<T>;
```

Comment:
With generic type when using `ReactIcon` it inherits `IconTypes`.

Reply:
I see what you mean but we might not need to inherit the `IconTypes` since the props are all controlled by the IconButton component.

I've updated the icon prop type to be `React.ElementType` and omitted `children` just like you did. I believe this should fix the issue. Thanks a lot